platforms/posix/src/px4_daemon/server.cpp

```diff
@@ -124,12 +124,30 @@ Server::_server_main(void *arg)
 
 		client_send_packet_s packet;
 
-		int bytes_read = read(client_send_pipe_fd, &packet, sizeof(packet));
+		// We only read as much as we need, otherwise we might get out of
+		// sync with packets.
+		int bytes_read = read(client_send_pipe_fd, &packet, sizeof(client_send_packet_s::header));
 
 		if (bytes_read > 0) {
-			_parse_client_send_packet(packet);
 
-		} else if (bytes_read == 0) {
+			// Using the header we can determine how big the payload is.
+			int payload_to_read = sizeof(packet)
+					      - sizeof(packet.header)
+					      - sizeof(packet.payload)
+					      + packet.header.payload_length;
```

Comment:
Isn't `sizeof(packet) - sizeof(packet.header) - sizeof(packet.payload)` equal to 0?

Reply:
Not necessarily if there are padding bytes between the header and the payload. Or at the struct end.

Reply:
Hm, ok are padding bytes counted in `sizeof`?

Reply:
Yes for the padding bytes that are inside the struct.